### PR TITLE
add readiness probe types

### LINKED ISSUE #405: update readiness probes to match http probe functionality
Add support for http and tcp probes

### PR DESCRIPTION
**Pull request by mangas**
Resolves #405

### CODE REVIEW

File: pkg/kev/kev_reconcile_test.go [hunk 1]
```diff
@@ -491,6 +491,18 @@ var _ = Describe("Reconcile", func() {
...
+				It("should have a valid http readiness probe", func() {
+					svcCfg, err := env.GetService("wordpress")
+					Expect(err).To(Succeed())
+					Expect(svcCfg.GetLabels()).To(
+						HaveKeyWithValue(config.LabelWorkloadReadinessProbeType, kubernetes.ProbeTypeHTTP.String()))
+					Expect(svcCfg.GetLabels()).To(
+						HaveKeyWithValue(config.LabelWorkloadReadinessProbeHTTPPort, "8080"))
+					Expect(svcCfg.GetLabels()).To(
+						HaveKeyWithValue(config.LabelWorkloadReadinessProbeHTTPPath, "/status"))
+					Expect(svcCfg.GetLabels()).NotTo(HaveKey(config.LabelWorkloadLivenessProbeCommand))
```
Comment (marcinc) [inline pkg/kev/kev_reconcile_test.go:504]: did you mean `config.LabelWorkloadReadinessProbeCommand`?

**Review by marcinc — COMMENTED** [review]
A few small comments. LG otherwise.

**Review by ezodude — COMMENTED** [review]
Feels like Liveness Probe DeJa Vu 😉 

Looks good, apart from the comments left by @marcinc

File: pkg/kev/kev_reconcile_test.go (hunk 1 above)
Comment (mangas) [reply pkg/kev/kev_reconcile_test.go]: copy paste leftover, good catch 🙏

File: pkg/kev/converter/kubernetes/project_service_test.go [hunk 2]
```diff
@@ -1539,6 +1539,123 @@ var _ = Describe("ProjectService", func() {
...
+
+			BeforeEach(func() {
+				labels = composego.Labels{
+					config.LabelWorkloadReadinessProbeType:     ProbeTypeHTTP.String(),
+					config.LabelWorkloadReadinessProbeHTTPPort: port,
+					config.LabelWorkloadReadinessProbeHTTPPath: path,
+				}
+			})
+
+			It("returns label value", func() {
```
[the remark this answers is not in the document]
Comment (mangas) [reply pkg/kev/converter/kubernetes/project_service_test.go]: updated

File: pkg/kev/converter/kubernetes/project_service_test.go [hunk 3]
```diff
@@ -1539,6 +1539,123 @@ var _ = Describe("ProjectService", func() {
...
+				pt, err := projectService.readinessProbeType()
+				Expect(err).To(Succeed())
+				Expect(*pt).To(Equal(ProbeTypeNone))
+				Expect(projectService.readinessProbeCommand()).To(Equal([]string{}))
+			})
+		})
+	})
+
+	Describe("readinessProbeHTTP", func() {
+		Context("and supplied as string port and parth", func() {
```
[the remark this answers is not in the document]
Comment (mangas) [reply pkg/kev/converter/kubernetes/project_service_test.go]: good catch, fixed

**Review by ezodude — APPROVED** [review]
💯

**Pull request merged**